packages/format-library/src/link/utils.js

```diff
@@ -111,3 +111,134 @@ export function createLinkFormat( { url, type, id, opensInNewWindow } ) {
 
 	return format;
 }
+
+/* eslint-disable jsdoc/no-undefined-types */
+/**
+ * Get the start and end boundaries of a given format from a rich text value.
+ *
+ *
+ * @param {RichTextValue} value      the rich text value to interrogate.
+ * @param {string}        format     the identifier for the target format (e.g. `core/link`, `core/bold`).
+ * @param {number?}       startIndex optional startIndex to seek from.
+ * @param {number?}       endIndex   optional endIndex to seek from.
+ * @return {Object}	object containing start and end values for the given format.
+ */
+/* eslint-enable jsdoc/no-undefined-types */
+export function getFormatBoundary(
+	value,
+	format,
+	startIndex = value.start,
+	endIndex = value.end
+) {
+	const EMPTY_BOUNDARIES = {
+		start: null,
+		end: null,
+	};
+
+	const { formats } = value;
+	let targetFormat;
+	let initialIndex;
+
+	if ( ! formats?.length ) {
+		return EMPTY_BOUNDARIES;
+	}
+
+	// Clone formats to avoid modifying source formats.
+	const newFormats = formats.slice();
+
+	const formatAtStart = find( newFormats[ startIndex ], {
+		type: format.type,
+	} );
+
+	const formatAtEnd = find( newFormats[ endIndex ], {
+		type: format.type,
+	} );
+
+	const formatAtEndMinusOne = find( newFormats[ endIndex - 1 ], {
+		type: format.type,
+	} );
+
+	if ( Boolean( formatAtStart ) ) {
```

Comment:
Using the `Boolean` approach here but I guess I could just as easily use the `!!` if that's more inline with current style guide.

```suggestion
	if ( !! formatAtStart  ) {
```

Reply:
I updated this.